arcgis-ios-sdk-samples/Maps/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningViewController.swift

```diff
@@ -0,0 +1,306 @@
+            self.presentAlert(error: error)
+        }
+    }
+    
+    /// A helper method to load one feature table at a time. Stop loading if
+    /// error occurs.
+    /// - Parameters:
+    ///   - tables: The feature tables to load.
+    ///   - completion: The load result for a table.
+    func loadTables<C>(_ tables: C, completion: @escaping (Result<Void, Error>) -> Void) where C: RandomAccessCollection, C.Element == AGSFeatureTable {
+        guard let table = tables.last else {
+            completion(.success(()))
+            return
+        }
+        table.load { [weak self] error in
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                self?.loadTables(tables.dropLast(), completion: completion)
+            }
+        }
+    }
+    
+    /// Display features on a certain floor level using definition expression.
+    /// - Parameter floor: The floor level of the features to be displayed.
+    func displayFeatures(onFloor floor: Int) {
+        (mapView.map!.operationalLayers as! [AGSLayer]).forEach { layer in
+            if layer.name == "Details" || layer.name == "Units" || layer.name == "Levels" {
+                if let featureLayer = layer as? AGSFeatureLayer {
+                    featureLayer.definitionExpression = "VERTICAL_ORDER = \(floor)"
+                }
+            }
+        }
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["ShowDeviceLocationUsingIndoorPositioningViewController"]
+    }
+    
+    deinit {
+        // Stop location display, which in turn stop the data source.
+        mapView.locationDisplay.stop()
+        // Reset indoors location data source.
+        indoorsLocationDataSource?.locationChangeHandlerDelegate = nil
+        indoorsLocationDataSource = nil
+        // Reset the API key.
+        AGSArcGISRuntimeEnvironment.apiKey = apiKey
+    }
+}
+
+// MARK: - AGSLocationChangeHandlerDelegate
+
+extension ShowDeviceLocationUsingIndoorPositioningViewController: AGSLocationChangeHandlerDelegate {
+    func locationDataSource(_ locationDataSource: AGSLocationDataSource, locationDidChange location: AGSLocation) {
+        // The floor level provided by the indoors beacons.
+        let floorText: String
+        if let floor = location.additionalSourceProperties[.floor] as? Int {
+            currentFloor = floor
+            floorText = String(format: "Floor level: %d", floor)
+        } else {
+            floorText = "Floor not available"
+        }
+        
+        // The horizontal accuracy of the positioning signal from the sensors.
+        let horizontalAccuracy = measurementFormatter.string(from: Measurement(value: Double(location.horizontalAccuracy), unit: UnitLength.meters))
```

Comment:
`location.horizontalAccuracy` is a `Double` value.

```suggestion
        let horizontalAccuracy = measurementFormatter.string(from: Measurement(value: location.horizontalAccuracy, unit: UnitLength.meters))
```